### ISSUE TITLE
Import Error Message When Successfully Creating New Item, Comments Not Importing

### ISSUE THREAD

**Issue by steveswinsburg** [sakaiproject/sakai#1386]
_From @kyleblythe on October 22, 2015 18:50_

When successfully creating a new item with scores/comments and clicking "Finish," the following error occurs:

![image](https://cloud.githubusercontent.com/assets/11447474/10674961/aa31108e-78cb-11e5-9576-efdbb80e6628.png)

Upon clicking "Finish" again, you get the following error message:

![image](https://cloud.githubusercontent.com/assets/11447474/10674997/d1374fae-78cb-11e5-9d11-8c26e1261ed3.png)

However, after resetting the tool, the item was indeed created (though without comments):

![image](https://cloud.githubusercontent.com/assets/11447474/10675054/1fa4f060-78cc-11e5-9775-befc2df3270f.png)

_Copied from original issue: steveswinsburg/gradebookNG#271_

**Comment by steveswinsburg** [comment]
Unclear about first error (need logs) but second error is because it actually did import and then clicking finish again caused the duplicate assignment.

In any case, the `ConflictingNameException` needs to be caught in our addAssignment service method.

**Comment by steveswinsburg** [comment]
_From @payten on October 22, 2015 21:8_

Looking at the logs, prior to the "You cannot save..." exception, there weren't any clear exceptions preceding. Of the two times the "You cannot save..." was logged, the first was preceded by:

```
2015-10-22 14:43:18,896  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: OVER_LIMIT, 422, tst363, 90, null
2015-10-22 14:43:19,039  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst116, null, null
2015-10-22 14:43:19,180  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst288, null, null
2015-10-22 14:43:19,319  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst283, null, null
2015-10-22 14:43:19,463  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst282, null, null
2015-10-22 14:43:19,602  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst101, null, null
2015-10-22 14:43:19,737  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst285, null, null
2015-10-22 14:43:19,873  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst286, null, null
2015-10-22 14:43:20,011  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst287, null, null
2015-10-22 14:43:20,150  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst284, null, null
2015-10-22 14:43:20,290  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst289, null, null
2015-10-22 14:43:20,431  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst322, null, null
2015-10-22 14:43:20,572  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst163, null, null
2015-10-22 14:43:20,711  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst571, null, null
2015-10-22 14:43:20,850  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst227, null, null
2015-10-22 14:43:20,988  INFO http-nio-8080-exec-61 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 422, tst102, null, null
```

The second was preceded by:

```
2015-10-22 14:45:27,108  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: OVER_LIMIT, 423, tst363, 80, null
2015-10-22 14:45:27,261  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst116, null, null
2015-10-22 14:45:27,451  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: OVER_LIMIT, 423, tst288, 60, null
2015-10-22 14:45:27,604  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst283, null, null
2015-10-22 14:45:27,761  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst282, null, null
2015-10-22 14:45:27,917  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst101, null, null
2015-10-22 14:45:28,075  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst285, null, null
2015-10-22 14:45:28,238  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst286, null, null
2015-10-22 14:45:28,391  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst287, null, null
2015-10-22 14:45:28,539  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst284, null, null
2015-10-22 14:45:28,690  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst289, null, null
2015-10-22 14:45:28,841  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst322, null, null
2015-10-22 14:45:29,004  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst163, null, null
2015-10-22 14:45:29,165  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst571, null, null
2015-10-22 14:45:29,312  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst227, null, null
2015-10-22 14:45:29,461  INFO http-nio-8080-exec-36 org.sakaiproject.gradebookng.tool.panels.importExport.GradeImportConfirmationStep - Saving grade: NO_CHANGE, 423, tst102, null, null
```

**Comment by steveswinsburg** [comment]
Hmm, it's still not clear about the 'You cannot save' error, maybe it tried to create two assignments with the same name... But the comments not saving is interesting, in the above there is no comment, so either there was none to start with or it hasn't detected it correctly? Sounds like this needs a bit more work to debug.

In addition, everything is in the onSubmit of the form which means the thread will just hang there until it's done. I think this might need to be made async which would fix #263 as well.

**Comment by kyleblythe** [comment]
This looks like an old issue that was already addressed. Can no longer reproduce this issue.
